api/v1alpha1/oadp_types.go

```diff
@@ -68,6 +67,191 @@ const TechPreviewAck UnsupportedImageKey = "tech-preview-ack"
+	ClientQPS *string `json:"client-qps,omitempty"`
+	// Maximum number of requests by the server to the Kubernetes API in a short period of time.
+	// +optional
+	ClientBurst *int `json:"client-burst,omitempty"`
+	// Page size of requests by the server to the Kubernetes API when listing objects during a backup. Set to 0 to disable paging.
+	// +optional
+	ClientPageSize *int `json:"client-page-size,omitempty"`
+	// The address to expose the pprof profiler.
+	// +optional
+	ProfilerAddress string `json:"profiler-address,omitempty"`
+	// How often (in nanoseconds) to check status on backup/restore operations after backup/restore processing.
+	// +optional
+	ItemOperationSyncFrequency *time.Duration `json:"item-operation-sync-frequency,omitempty"`
+	// The format for log output. Valid values are text, json. (default text)
+	// +kubebuilder:validation:Enum=text;json
+	// +optional
+	FormatFlag string `json:"log-format,omitempty"`
+	// How often (in nanoseconds) 'maintain' is run for backup repositories by default.
+	// +optional
+	RepoMaintenanceFrequency *time.Duration `json:"default-repo-maintain-frequency,omitempty"`
+	// How often (in nanoseconds) garbage collection checks for expired backups. (default is 1 hour)
+	// +optional
+	GarbageCollectionFrequency *time.Duration `json:"garbage-collection-frequency,omitempty"`
+	// Backup all volumes with pod volume file system backup by default.
+	// +optional
+	DefaultVolumesToFsBackup *bool `json:"default-volumes-to-fs-backup,omitempty"`
+	// How long (in nanoseconds) to wait on asynchronous BackupItemActions and RestoreItemActions to complete before timing out. (default is 1 hour)
+	DefaultItemOperationTimeout *time.Duration `json:"default-item-operation-timeout,omitempty"`
+	// How long (in nanoseconds) to wait for resource processes which are not covered by other specific timeout parameters. (default is 10 minutes)
+	ResourceTimeout *time.Duration `json:"resource-timeout,omitempty"`
+	// Max concurrent connections number that Velero can create with kube-apiserver. Default is 30. (default 30)
+	MaxConcurrentK8SConnections *int `json:"max-concurrent-k8s-connections,omitempty"`
+}
+
+// GlobalFlags are flags that are defined across Velero CLI commands
+type GlobalFlags struct {
+	// We use same namespace as DataProtectionApplication
+	// Namespace string `json:"namespace,omitempty"`
+
+	// Features is an existing field in DataProtectionApplication
+	// --kubebuilder:validation:Enum=EnableCSI;EnableAPIGroupVersions;EnableUploadProgress
+	// Features  []VeleroFeatureFlag `json:"features,omitempty"`
+
+	// Show colored output in TTY
+	// +optional
+	Colorized *bool `json:"colorized,omitempty"`
+	// Boolean flags. Not handled atomically because the flag.Value interface
+	// does not let us avoid the =true, and that shorthand is necessary for
+	// compatibility. TODO: does this matter enough to fix? Seems unlikely.
+	// +optional
+	ToStderr *bool `json:"logtostderr,omitempty"` // The -logtostderr flag.
```

Comment:
We are losing context where things came from.

Like that this came from https://github.com/openshift/velero/blob/240b4e666fe15ef98defa2b51483fe87ac9996fb/pkg/cmd/velero/velero.go#L125

Prior code had these in comments. I would add these comments back in.

Unless this refactor is meant to be short lived, I am in favor of having a map of where things came from.

Reply:
Added back info